src/lib/operators/jit_operator/jit_operations.hpp

```diff
@@ -0,0 +1,195 @@
+    using ResultType = decltype(op_func(typed_lhs, typed_rhs));
+    // This templated function returns the JitDataType enum value for a given ResultType.
+    return jit_data_type<ResultType>();
+  };
+
+  const auto catching_func = InvalidTypeCatcher<decltype(func), JitDataType>(func);
+
+  // The type information from the lhs and rhs are combined into a single value for dispatching without nesting.
+  const auto combined_types = static_cast<uint8_t>(lhs) << 8 | static_cast<uint8_t>(rhs);
+  switch (combined_types) {
+    BOOST_PP_SEQ_FOR_EACH_PRODUCT(JIT_COMPUTE_TYPE_CASE, (JIT_DATA_TYPE_INFO)(JIT_DATA_TYPE_INFO))
+    default:
+      Fail("unreachable");
+  }
+}
+
+__attribute__((noinline)) void jit_not(const JitMaterializedValue& lhs, JitMaterializedValue& result) {
+  DebugAssert(lhs.data_type() == JitDataType::Bool && result.data_type() == JitDataType::Bool,
+              "invalid type for operation");
+  result.is_null() = lhs.is_null();
+  result.as<uint8_t>() = !lhs.as<uint8_t>();
+}
+
+__attribute__((noinline)) void jit_and(const JitMaterializedValue& lhs, const JitMaterializedValue& rhs,
+                                       JitMaterializedValue& result) {
+  DebugAssert(lhs.data_type() == JitDataType::Bool && rhs.data_type() == JitDataType::Bool &&
+                  result.data_type() == JitDataType::Bool,
+              "invalid type for operation");
+
+  // three-valued logic AND
+  if (lhs.is_null()) {
+    result.as<uint8_t>() = false;
+    result.is_null() = rhs.is_null() || rhs.as<uint8_t>();
+  } else {
+    result.as<uint8_t>() = lhs.as<uint8_t>() && rhs.as<uint8_t>();
+    result.is_null() = lhs.as<uint8_t>() && rhs.is_null();
+  }
+}
+
+__attribute__((noinline)) void jit_or(const JitMaterializedValue& lhs, const JitMaterializedValue& rhs,
+                                      JitMaterializedValue& result) {
+  DebugAssert(lhs.data_type() == JitDataType::Bool && rhs.data_type() == JitDataType::Bool &&
+                  result.data_type() == JitDataType::Bool,
+              "invalid type for operation");
+
+  // three-valued logic OR
+  if (lhs.is_null()) {
+    result.as<uint8_t>() = true;
+    result.is_null() = rhs.is_null() || !rhs.as<uint8_t>();
+  } else {
+    result.as<uint8_t>() = lhs.as<uint8_t>() || rhs.as<uint8_t>();
+    result.is_null() = !lhs.as<uint8_t>() && rhs.is_null();
+  }
+}
+
+__attribute__((noinline)) void jit_is_null(const JitMaterializedValue& lhs, JitMaterializedValue& result) {
+  DebugAssert(result.data_type() == JitDataType::Bool, "invalid type for operation");
+
+  result.is_null() = false;
+  result.as<uint8_t>() = lhs.is_null();
+}
+
+__attribute__((noinline)) void jit_is_not_null(const JitMaterializedValue& lhs, JitMaterializedValue& result) {
+  DebugAssert(result.data_type() == JitDataType::Bool, "invalid type for operation");
+
+  result.is_null() = false;
+  result.as<uint8_t>() = !lhs.is_null();
```

Comment:
Would `result.set<uint8_t>(...)` also work?

Reply:
Yes! Changed it accordingly.